src/app/OperationalSessionSetup.cpp

```diff
@@ -191,27 +191,29 @@ void OperationalSessionSetup::UpdateDeviceData(const Transport::PeerAddress & ad
     if (mState == State::ResolvingAddress)
     {
         MoveToState(State::HasAddress);
-        err = EstablishConnection();
```

Comment:
Leaving a comment here so that I can get a chance to review this first thing tomorrow morning!